### PR TITLE
Html task

### PR DESCRIPTION
**Pull request by gauntface**
This adds a task to manage html pages in the build process.

Major thing this does is:

- Minify HTML
- Inline CSS and JS if the tags have an 'inline' attribute

Things this doesn't currently do (and not sure if we want to do this) - minify / optimise inline styles and javascript.

@addyosmani @surma + anyone else who has any ideas for improving this / anything we might be missing.

### CODE REVIEW

File: src/src/index.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <meta id="theme-color" name="theme-color" content="#FEBC04">
+    <title>Web Starter Kit</title>
+
+    <link rel="manifest" href="/manifest.json">
+    <link rel="icon" href="/images/icons/icon-192x192.png" sizes="192x192" type="image/png">
+    <link rel="icon" href="/images/icons/icon-384x384.png" sizes="384x384" type="image/png">
+
+    <!-- By adding the 'inline' attribute, the build process will inline
+    this CSS file to improve load time performance-->
+    <link rel="stylesheet" href="/styles/index.css" type="text/css" inline>
```
Comment (surma) [inline src/src/index.html]: To be 100% compliant, should we use `data-inline`? Or is `inline` an official attribute?

**Comment by surma** [comment]
Only one minor nit and I am not even sure we can do anything about that.

Also tests.

File: src/src/index.html (hunk 1 above)
Comment (gauntface) [inline src/src/index.html]: We can customise the attribute, so data-inline sounds like a great idea tbh

**Comment by Garbee** [comment]
The auto-inlining clashes with work in PR #823 from another contributor. Maybe merge that in and then do an upgrade to the implementation?

**Comment by surma** [comment]
@Garbee This branch is a completely new tree (orphan). No clashing will ever happen :)

**Comment by Garbee** [comment]
oh, just noticed going into a 2.x branch. Yea nvm then.

**Comment by gauntface** [comment]
BUT . . . . . @Garbee thanks for bringing it to me attention - was a great feature to add in.

**Comment by gauntface** [comment]
@surma happy with the current state?

I'm seeing some flakiness on the html watch task - will have to keep an eye on it.

**Pull request merged**